wallet-new/src/Cardano/Wallet/WalletLayer.hs

```diff
@@ -563,3 +572,19 @@ instance Buildable RedeemAdaError where
         bprint ("RedeemAdaWalletIdDecodingFailed " % build) txt
     build (RedeemAdaInvalidRedemptionCode txt) =
         bprint ("RedeemAdaInvalidRedemptionCode " % build) txt
+
+data NewTransactionError =
```

Comment:
This `NewTransactionError` is confusing, because we already have another error type called `NewPaymentError`, if I recall correctly. I suggest either to extend that one OR rename this one `NewUnsignedTransactionError`, to make really explicit which API of the wallet layer we are calling.